modules/aruco/samples/detect_markers.cpp

```diff
@@ -0,0 +1,234 @@
+    for(int i = 0; i < argc && idx == -1; i++)
+        if(string(argv[i]) == param) idx = i;
+    if(idx == -1 || (idx + 1) >= argc)
+        return defvalue;
+    else
+        return argv[idx + 1];
+}
+
+
+
+/**
+ */
+static bool readCameraParameters(string filename, Mat &camMatrix, Mat &distCoeffs) {
+    FileStorage fs(filename, FileStorage::READ);
+    if(!fs.isOpened())
+        return false;
+    fs["camera_matrix"] >> camMatrix;
+    fs["distortion_coefficients"] >> distCoeffs;
+    return true;
+}
+
+
+
+/**
+ */
+static bool readDetectorParameters(string filename, aruco::DetectorParameters &params) {
+    FileStorage fs(filename, FileStorage::READ);
+    if(!fs.isOpened())
+        return false;
+    fs["adaptiveThreshWinSizeMin"] >> params.adaptiveThreshWinSizeMin;
+    fs["adaptiveThreshWinSizeMax"] >> params.adaptiveThreshWinSizeMax;
+    fs["adaptiveThreshWinSizeStep"] >> params.adaptiveThreshWinSizeStep;
+    fs["adaptiveThreshConstant"] >> params.adaptiveThreshConstant;
+    fs["minMarkerPerimeterRate"] >> params.minMarkerPerimeterRate;
+    fs["maxMarkerPerimeterRate"] >> params.maxMarkerPerimeterRate;
+    fs["polygonalApproxAccuracyRate"] >> params.polygonalApproxAccuracyRate;
+    fs["minCornerDistanceRate"] >> params.minCornerDistanceRate;
+    fs["minDistanceToBorder"] >> params.minDistanceToBorder;
+    fs["minMarkerDistanceRate"] >> params.minMarkerDistanceRate;
+    fs["doCornerRefinement"] >> params.doCornerRefinement;
+    fs["cornerRefinementWinSize"] >> params.cornerRefinementWinSize;
+    fs["cornerRefinementMaxIterations"] >> params.cornerRefinementMaxIterations;
+    fs["cornerRefinementMinAccuracy"] >> params.cornerRefinementMinAccuracy;
+    fs["markerBorderBits"] >> params.markerBorderBits;
+    fs["perspectiveRemovePixelPerCell"] >> params.perspectiveRemovePixelPerCell;
+    fs["perspectiveRemoveIgnoredMarginPerCell"] >> params.perspectiveRemoveIgnoredMarginPerCell;
+    fs["maxErroneousBitsInBorderRate"] >> params.maxErroneousBitsInBorderRate;
+    fs["minOtsuStdDev"] >> params.minOtsuStdDev;
+    fs["errorCorrectionRate"] >> params.errorCorrectionRate;
+    return true;
+}
+
+
+
+/**
+ */
+int main(int argc, char *argv[]) {
+
+    if(!isParam("-d", argc, argv)) {
+        help();
+        return 0;
+    }
+
+    int dictionaryId = atoi(getParam("-d", argc, argv).c_str());
+    aruco::Dictionary dictionary =
+        aruco::getPredefinedDictionary(aruco::PREDEFINED_DICTIONARY_NAME(dictionaryId));
+
+    bool showRejected = false;
+    if(isParam("-r", argc, argv)) showRejected = true;
+
+    bool estimatePose = false;
+    Mat camMatrix, distCoeffs;
+    if(isParam("-c", argc, argv)) {
+        bool readOk = readCameraParameters(getParam("-c", argc, argv), camMatrix, distCoeffs);
+        if(!readOk) {
+            cerr << "Invalid camera file" << endl;
+            return 0;
+        }
+        estimatePose = true;
+    }
+    float markerLength = (float)atof(getParam("-l", argc, argv, "0.1").c_str());
```

Comment:
I don't think it is a good idea to leave it open to any units because the consequence of the Tvec array being in the same units is hidden. This might create confusion. I would prefer choosing meters and mentioning that in help and in documentation. That way there is clarity.